### PR TITLE
iRegs: refactor wayfinder to fix customizations and BEM

### PR DESCRIPTION
**Pull request by anselmbradford**
The iRegs wayfinder includes its own component organism `o-regulations-wayfinder`. Inside this component, the layout is handled by `content_wrapper`, `content_main`, and `content_sidebar` to align the left/right columns with the rest of the page layout. However, these classes have styles specific to the page layout, so `o-regulations-wayfinder` overrides styles of these classes, which it should not do (outside of margin adjustments). Additionally, it mixes in `content__flush-top content__flush-bottom`, which is fine, but the combined mixing of utility classes and overrides makes this brittle. 

This PR rewrites the wayfinder to handle its own layout and not depend on external layout classes.

Additionally, the wayfinder contained two classes `o-regulations-wayfinder__content` and `o-regulations-wayfinder__version`, which are written as if they were modifiers in our flavor of BEM, but are actually treated as BEM elements. An additional class of this type, `o-regulations-wayfinder__heading`, was unused and was removed.

## Removals

- Remove `o-regulations-wayfinder__heading`

## Changes

- Refactors wayfinder to use `o-regulations-wayfinder_main` and `o-regulations-wayfinder_sidebar` in place of customized `content_main` and `content_sidebar`.

## How to test this PR

1. Visit a regulation, like http://localhost:8000/rules-policy/regulations/1002/2/ and scroll down till the wayfinder shows. It should appear like production. Resize to tablet and mobile.

## Screenshots
Screen Shot 2023-05-19 at 10 29 33 AM
Screen Shot 2023-05-19 at 10 29 42 AM
Screen Shot 2023-05-19 at 10 44 34 AM

### CODE REVIEW

**Review by wpears — COMMENTED** [review]
This appears busted for me on the desktop view:
Screen Shot 2023-05-19 at 3 52 49 PM

**Comment by anselmbradford** [comment]
Thanks @wpears tried to remove the wrapper, but you're right it won't work. Fixed in https://github.com/cfpb/consumerfinance.gov/pull/7757/commits/81f1c01e755d97668a9117adfd30fbae06832e84

However, this will be slightly misaligned

Screen Shot 2023-05-19 at 4 19 47 PM

This is because our 1 to 3 layout is not actually a correct 1 to 3 ratio. This will be fixed in https://github.com/cfpb/consumerfinance.gov/pull/7754, which also contains the changes in this PR.

We could wait, or merge this to make 7754 easier to review and live with the misaligned divider for a hopefully brief window.

File: cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less [hunk 1]
```diff
@@ -28,35 +30,49 @@
...
 
-  .content_sidebar {
-    display: none;
-    .respond-to-min( @bp-med-min, {
+  // Desktop
+  .respond-to-min( @bp-med-min, {
+    &_wrapper {
+      display: grid;
+      grid-template-columns: 1fr 3fr;
+      max-width: 1170px;
```
Comment (wpears) [inline cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less:48]:
```suggestion
      max-width: 1260px;
```

Comment (anselmbradford) [reply cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less:48]: ?

File: cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less [hunk 2]
```diff
@@ -28,35 +30,49 @@
...
+      margin: 0 auto;
+    }
+
+    &_sidebar {
       display: inline-block;
-    } );
-  }
+      position: relative;
```
Comment (wpears) [inline cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less:54]:
```suggestion
      position: relative;
      padding-left: unit(45px / @base-font-size-px, rem);
```

Comment (anselmbradford) [reply cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less:54]: Are you trying to fix the dividing line alignment? I should be fixed in https://github.com/cfpb/consumerfinance.gov/pull/7754

File: cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less (hunk 1 above)
Comment (wpears) [reply cfgov/unprocessed/apps/regulations3k/css/reg-wayfinder.less:48]: oh, I hadn't seen your response! This fixes the misalignment, but the other correction is better.

**Comment by anselmbradford** [comment]
Oops I actually was going to keep this open till 7754 was signed off

**Pull request merged**